virtualfish/virtual.fish

```diff
@@ -121,6 +121,46 @@ function __vf_deactivate --description "Deactivate this virtualenv"
     set -e VIRTUAL_ENV
 end
 
+function __vfsupport_find_python --description "Search for and return Python path"
+    set -l python_arg $argv[1]
+    set -l py_path (which $python_arg)
+    set -l py_version (string replace "python" "" $python_arg)
+    set -l brew_path "/usr/local/opt/python@$py_version/bin/python$py_version"
+    # Executable on PATH (python3/python3.8) or full interpreter path
+    if test -x "$py_path"
```

Comment:
If you use `command -s`, you don't actually need to test that it's executable, fish will take care of it. I'm pretty sure most `which` implementations would as well.

Also if nothing is found it will return false.

So this can be

```fish
if set -l py_path (command -s python)
    set python "$py_path"
```

and the $py_path variable doesn't have to appear anywhere else (so you can ditch the `set` above!)